odoo/addons/base/models/ir_http.py

```diff
@@ -315,19 +315,20 @@ def _binary_record_content(
         field_def = record._fields[field]
         if field_def.type == 'binary' and field_def.attachment and not field_def.related:
             if model != 'ir.attachment':
-                field_attachment = self.env['ir.attachment'].sudo().search_read(domain=[('res_model', '=', model), ('res_id', '=', record.id), ('res_field', '=', field)], fields=['raw', 'mimetype', 'checksum'], limit=1)
+                Attachment_sudo = self.env['ir.attachment'].sudo()
+                field_attachment = Attachment_sudo.search_read(domain=[('res_model', '=', model), ('res_id', '=', record.id), ('res_field', '=', field)], fields=['mimetype', 'checksum', 'raw'], limit=1)
                 if field_attachment:
                     mimetype = field_attachment[0]['mimetype']
                     content = field_attachment[0]['raw']
                     filehash = field_attachment[0]['checksum']
             else:
                 mimetype = record['mimetype']
-                content = record['raw']
+                content = record.get_lazy_raw()
```

Comment:
Hmm, on the risk impact scale, changing the `ir.attachment` implementation is *way* higher than perturbing the website APIs by adding extra parameters for the xaccel special cases.

One way to pass information to called methods without modifying their parameters is to put it in the `request.context` (cfr examples in our codebase). That might be a way to convey the "xaccel support status" in your case. It's generally considered a dirty way of doing things, but when you're specifically trying to avoid modifying method parameters, it could do the job.

Reply:
Those were my thoughts while working on it, it does seem like a huge change just to skip passing a parameter. I will try the request.context solution.

Reply:
@odony I am waiting for some changes that are currently being built by @Gorash (I think?) for lazy loading. Combining those with the xaccel changes should provide the ideal infrastructure for this (lazy loading + discard reading the data if xaccel header is detected before serving)